docs-src/concepts.md

```diff
@@ -343,6 +343,30 @@ status:
     Message: "foobar" is an FooBar.
 ```
 
+### GatewayClass controller selection
+
+The `GatewayClass.spec.controller` field is used to determine whether
+or not a given `GatewayClass` is managed by the controller.
+This format of this field is opaque and specific to a particular controller.
+Which GatewayClass is selected by a given controller field depends on how
+various controller(s) in the cluster interpret this field.
+
+It is RECOMMENDED that controller authors/deployments make their
+selection unique by using a domain / path combination under their
+administrative control (e.g. controller managing of all `controller`s
+starting with `acme.io` is the owner of the `acme.io` domain) to avoid
+conflicts.
+
+Controller versioning can be done by encoding the version of a
+controller into the path portion. An example scheme could be (similar
+to container URIs):
+
+```text
+acme.io/gateway:1.14   // Use version 1.14
+acme.io/gateway:1.15   // Use version 1.15
```

Comment:
I believe these would be invalid names if we were using the same validation as Ingress. Each path segment there must conform relatively strict validation: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L177.

Reply:
Do we do validations on the Ingress class name?

Reply:
Yeah our validations are quite strict on Ingress and really and k8s resources. It won't be quite as straightforward to do that here until we have a webhook. Until then, I'm hesitant to suggest patterns that would be invalid for IngressClass - ideally we'd want the same controller names to be valid for both. As a slight variation of this, maybe we could suggest `acme.io/gateway/v1` and `acme.io/gateway/v2`? I believe that would also be a valid IngressClass controller name. (That validation relies on the [IsDomainPrefixedPath](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L131) validation function).

Reply:
+1 on showing a validation safe example. Best to mention that future spec releases will validate this field to the a HTTP URL.

Reply:
TIL and makes sense. Updated, PTAL.